### DISCUSSION TITLE
Watermeter - Jump in m3

### DISCUSSION THREAD

**Discussion by malzmandl** [jomjol/AI-on-the-edge-device#1622] [General] [answered]
Dear all, 
I noticed one issue with my watermeter. The m3 count shows already e.g. 493 although the real value would be 492.6. The raw value returned jumps from 492.60 to 493.61 which leads to the error: Rate too high. 
Is there a setting which might help? 

Data viewer:
![image](https://user-images.githubusercontent.com/52219134/208302720-75f27f4e-10ec-4b6c-a361-4de191170ec7.png)

Please find below the recongnition image (meanwhile the correct value would be 492.8619 but the raw value is reported as 493.8619)
![image](https://user-images.githubusercontent.com/52219134/208303220-911952d1-9103-46dc-a103-0fcaffd068ea.png)

I definitly don't blame the system as it really looks like 493.6 ;-) Based on the context that the last correct value was 492.6 a reading of 493.65 doesn't make sense. Anything I can change to fix this? 

Thank you in advance.

PS: Thanks for the great project! Really amazing!!

**Comment by FajnB** [comment]
I have noticed the same issue: https://github.com/jomjol/AI-on-the-edge-device/discussions/1611

It also seems to be directly connected to issue: https://github.com/jomjol/AI-on-the-edge-device/discussions/626 but I am running version **v13.0.7** and have set the CkeckDigitIncreaseConsistency to true. Only thing which might be off is that my algorithm is set to dig-cont_0600_s3.tflite

**Reply by ghost** [reply]
> It also seems to be directly connected to issue: #626 but I am running version **v13.0.7** and have set the CkeckDigitIncreaseConsistency to true. Only thing which might be off is that my algorithm is set to dig-cont_0600_s3.tflite

CkeckDigitIncreaseConsistency is for class11 only. So it can not change anything in combination with dig-cont or dig-class100.

**Comment by FajnB** [comment]
The problem seems to go away if `dig-class11_1500_s2.tflite` algorithm is used and the `CkeckDigitIncreaseConsistency` is set to `True`.

So far it did flip the m<sup>3</sup> to another digit when 0.9999 m<sup>3</sup> went to another m<sup>3</sup>.
i.e. The EPS detected start of new m<sup>3</sup> on 1.0000 m<sup>3</sup> and not on (in my case) 0.5000 m<sup>3</sup> where the water meter itself changed the number.

I will post an update on next change of counter if it still works good.

**Reply by malzmandl** [reply]
For me this is still not solving the issue to 100%. Sometimes everything works really well. Sometimes it still jumps.

**Comment by ghost** [comment, marked as answer]
> Dear all, I noticed one issue with my watermeter. The m3 count shows already e.g. 493 although the real value would be 492.6. The raw value returned jumps from 492.60 to 493.61 which leads to the error: Rate too high. Is there a setting which might help?
> 
> Data viewer: ![image](https://user-images.githubusercontent.com/52219134/208302720-75f27f4e-10ec-4b6c-a361-4de191170ec7.png)
> 
> Please find below the recongnition image (meanwhile the correct value would be 492.8619 but the raw value is reported as 493.8619) ![image](https://user-images.githubusercontent.com/52219134/208303220-911952d1-9103-46dc-a103-0fcaffd068ea.png)
> 
> I definitly don't blame the system as it really looks like 493.6 ;-) Based on the context that the last correct value was 492.6 a reading of 493.65 doesn't make sense. Anything I can change to fix this?
> 
> Thank you in advance.
> 
> PS: Thanks for the great project! Really amazing!!

Look at https://github.com/jomjol/AI-on-the-edge-device/wiki/ROI-Configuration#dig-class100--dig-cont-configuration. The bottom of the ROIs is to low.

**Reply by malzmandl** [reply]
Thank you. I have updated the alignment. Now it seems to work at least 90% of the times.
